consul/resource_consul_service_test.go

```diff
@@ -171,6 +205,90 @@ resource "consul_service" "example" {
 }
 `
 
+const testAccConsulServiceCheck = `
+resource "consul_node" "compute" {
+	name    = "compute-example"
+	address = "www.hashicorptest.com"
+}
+
+resource "consul_service" "example" {
+	name = "example"
+	node = "${consul_node.compute.name}"
+	port = 80
+
+	check {
+		check_id = "service:redis1"
+		name = "Redis health check"
+		notes = "Script based health check"
+		status = "passing"
+
+		definition {
```

Comment:
I basically think we should not have "definition" here and just allow specification under the `check {}` block, even though the API has definition on read. Does that make sense? Here's an example for how someone could write a check:


```
check {
    id = "service:redis1"
    name = "Redis health check"
    notes = "Script based health check"
    status = "passing"
    tls_skip_verify = false
    method = "PUT"
    interval = "5s"
    timeout = "1s"
    deregister_critical_service_after = "30s"
}
```

Reply:
This just seems cleaner and "more obvious" to me. It mirrors what you would expect I think writing check definitions in HCL or JSON.

Reply:
Yes, nesting doesn't add any value.